kotlinx-coroutines-core/common/src/Await.kt

```diff
@@ -11,11 +11,10 @@ import kotlin.coroutines.*
  * This function is **not** equivalent to `deferreds.map { it.await() }` which fails only when it sequentially
  * gets to wait for the failing deferred, while this `awaitAll` fails immediately as soon as any of the deferreds fail.
  *
- * This suspending function is cancellable.
- * If the [Job] of the current coroutine is cancelled or completed while this suspending function is waiting,
- * this function immediately resumes with [CancellationException].
- * There is a **prompt cancellation guarantee**. If the job was cancelled while this function was
- * suspended, it will not resume successfully. See [suspendCancellableCoroutine] documentation for low-level details.
+ * This suspending function is cancellable: if the [Job] of the current coroutine is cancelled or completed while this
```

Comment:
W.r.t. `This suspending function is cancellable`, this particular part:

>if the [Job] of the current coroutine is cancelled **or completed** while this suspending function is waiting, this function immediately resumes with [CancellationException].

Maybe I'm missing something, but isn't this scenario (Job is completed, but in its body the code is suspended and is waiting) impossible?

Reply:
Yeah, I simply kept it the way it was without thinking, but now I have no idea what the text was trying to convey. I replaced `is cancelled or completed while` with `is cancelled while` throughout the project, even outside the prompt cancellation explanations.